src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -351,12 +383,25 @@ BuildDistinctPlan(Query *masterQuery, Plan *subPlan)
 		column->varno = OUTER_VAR;
 	}
 
+	foreach(targetCell, targetList)
```

Comment:
Can we use `HasDistinctAggregate()` here?

Reply:
yes we can